frontend/src/pages/projects/types.ts

```diff
@@ -105,10 +105,7 @@ export type DataConnectionAWS = {
   data: AWSSecretKind;
 };
 
-export type DataConnection = {
-  type: DataConnectionType;
-  data: Record<string, unknown>; // likely will be a unified CR at some point
-} & DataConnectionAWS;
+export type DataConnection = DataConnectionAWS;
```

Comment:
We talked about this in slack.

This might make more sense until we have another valid data connection type:
```
export enum DataConnectionType {
  UNKNOWN = -1,
  AWS,
}

export type DataConnectionAWS = {
  type: DataConnectionType.AWS;
  data: AWSSecretKind;
};

export type DataConnection =
  | {
      type: DataConnectionType;
      data: K8sResourceCommon & {
        metadata: {
          name: string;
          namespace: string;
        };
      };
    }
  | DataConnectionAWS;
```